scripts/nbconverted/integrate-compression-hetnet.py

```diff
@@ -0,0 +1,336 @@
+# 
+# ## Datasets
+# 
+# 1. [MSigDb](https://doi.org/10.1073/pnas.0506580102 "Gene set enrichment analysis: A knowledge-based approach for interpreting genome-wide expression profiles") - Curated genesets that represent various biological processes
+# 2. [xCell](https://doi.org/10.1186/s13059-017-1349-1, "xCell: digitally portraying the tissue cellular heterogeneity landscape") - Curated genesets that describe profiles of different cell-types
+
+# In[1]:
+
+
+import os
+import csv
+import pandas as pd
+import seaborn as sns
+
+import hetio.hetnet
+import hetio.readwrite
+import hetio.stats
+
+
+# In[2]:
+
+
+get_ipython().run_line_magic('matplotlib', 'inline')
+
+
+# ## Define the metagraph and instantiate the graph
+
+# In[3]:
+
+
+kind_to_abbev = {
+    
+    # metanodes
+    'Gene': 'G',
+
+    # MSigDB Nodes
+    'Cancer-Hallmarks': 'H',
+    'Positional-Gene-Sets': 'C1',
+    'Curated-Gene-Sets-CPG': 'C2-CPG',
+    'Curated-Gene-Sets-REACTOME': 'C2-CP-REACTOME',
+    'Motif-Gene-Sets-MIR': 'C3-MIR',
+    'Motif-Gene-Sets-TFT': 'C3-TFT',
+    'Computational-Gene-Sets-CGN': 'C4-CGN',
+    'Computational-Gene-Sets-CM': 'C4-CM',
+    'GO-Gene-Sets-BP': 'C5-BP',
+    'GO-Gene-Sets-CC': 'C5-CC',
+    'GO-Gene-Sets-MF': 'C5-MF',
+    'Oncogenic-Gene-Sets': 'C6',
+    'Immunologic-Gene-Sets': 'C7',
+    
+    # xCell Nodes
+    'xCell-Cell-Type': 'XCELL',
+    
+    # metaedges
+    'participates': 'p',
+}
+
+metaedge_tuples = [
+    # MSigDB metaedges
+    ('Gene', 'Cancer-Hallmarks', 'participates', 'both'),
+    ('Gene', 'Positional-Gene-Sets', 'participates', 'both'),
+    ('Gene', 'Curated-Gene-Sets-CPG', 'participates', 'both'),
+    ('Gene', 'Curated-Gene-Sets-REACTOME', 'participates', 'both'),
+    ('Gene', 'Motif-Gene-Sets-MIR', 'participates', 'both'),
+    ('Gene', 'Motif-Gene-Sets-TFT', 'participates', 'both'),
+    ('Gene', 'Computational-Gene-Sets-CGN', 'participates', 'both'),
+    ('Gene', 'Computational-Gene-Sets-CM', 'participates', 'both'),
+    ('Gene', 'GO-Gene-Sets-BP', 'participates', 'both'),
+    ('Gene', 'GO-Gene-Sets-CC', 'participates', 'both'),
+    ('Gene', 'GO-Gene-Sets-MF', 'participates', 'both'),
+    ('Gene', 'Oncogenic-Gene-Sets', 'participates', 'both'),
+    ('Gene', 'Immunologic-Gene-Sets', 'participates', 'both'),
+    
+    # xCell metaedges
+    ('Gene', 'xCell-Cell-Type', 'participates', 'both'),
+]
+
+
+# In[4]:
+
+
+# Initialize the graph
+metagraph = hetio.hetnet.MetaGraph.from_edge_tuples(metaedge_tuples, kind_to_abbev)
+graph = hetio.hetnet.Graph(metagraph)
+
+
+# ## Gene Nodes
+
+# In[5]:
+
+
+# Load curated gene names from versioned resource 
+commit = '721204091a96e55de6dcad165d6d8265e67e2a48'
+url = 'https://raw.githubusercontent.com/cognoma/genes/{}/data/genes.tsv'.format(commit)
+gene_df = pd.read_table(url)
+
+# Only consider protein-coding genes
+gene_df = (
+    gene_df.query("gene_type == 'protein-coding'")
+    .drop_duplicates('entrez_gene_id')
```

Comment:
There should be no duplicate `entrez_gene_id` in this file. If worried an assert would be best.